File: max_area_brute_force.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        :time complexity: O(n^2)
        :space complexity:O(1)
        """
        if len(height)==0 or height is None:
            return 0
        maxx=0
        n=len(height)
        
        for i in range(n-1):
            for j in range(i+1, n):
                maxx = max(maxx, (j-i)*min(height[i],height[j]))
        
        return maxx
        

File: maxAreaBruteForce.java
/**
 * time complexity is O(n^2)
 * space complexity is O(1)
 * 
 * time limit exceeds on leetcode
 */
class Solution {
    public int maxArea(int[] height) {
        
        int len = height.length;
        int max = 0;
        for(int i = 0; i < len; i++) {
            for(int j = i+1; j < len; j++) {
                int area = (j-i)*Math.min(height[i], height[j]);
                max = Math.max(max, area);
            }
        }
        return max;
    }
}